domain/src/main/java/com/teamagam/gimelgimel/domain/messages/SendGeoMessageInteractor.java

```diff
@@ -16,34 +21,45 @@
 public class SendGeoMessageInteractor extends SendMessageInteractor<MessageGeo> {
 
     private final String mMessageText;
-    private final PointGeometry mMessageGeometry;
+    private final Geometry mMessageGeometry;
     private final String mMessageType;
 
+    private GeoEntity mGeoEntity;
+
     SendGeoMessageInteractor(
             @Provided ThreadExecutor threadExecutor,
             @Provided UserPreferencesRepository userPreferences,
             @Provided MessagesRepository messagesRepository,
             @Provided MessageNotifications messageNotifications,
             String text,
-            PointGeometry pg,
+            Geometry geometry,
             String type) {
         super(threadExecutor, userPreferences, messageNotifications, messagesRepository);
         mMessageText = text;
-        mMessageGeometry = pg;
+        mMessageGeometry = geometry;
         mMessageType = type;
     }
 
     @Override
     protected MessageGeo createMessage(String senderId) {
-        return new MessageGeo(null, senderId, null, createGeoEntity());
+        mMessageGeometry.accept(new CreateGeoEntityVisitor());
+        return new MessageGeo(null, senderId, null, mGeoEntity);
     }
 
-    private GeoEntity createGeoEntity() {
-        PointSymbol symbol = createSymbolFromType(mMessageType);
-        return new PointEntity("not_used", mMessageText, mMessageGeometry, symbol);
-    }
+    private class CreateGeoEntityVisitor implements IGeometryVisitor {
```

Comment:
Using the containing class field to store the visitor results couples the inner class to its containing class.
As of now we can't extract this inner-class to another file.
This can be resolved with a field and a getter for the result (within the inner class).